src/main/java/cz/smarteon/loxone/LoxoneCrypto.java

```diff
@@ -165,9 +167,9 @@ static String decrypt(final String data, final SecretKey sharedKey, final byte[]
             final Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
             final IvParameterSpec ivspec = new IvParameterSpec(sharedKeyIv);
             cipher.init(Cipher.DECRYPT_MODE, sharedKey, ivspec);
-            return new String(cipher.doFinal(Base64.decode(data)));
+            return new String(cipher.doFinal(base64ToBytes(data)));
         } catch (NoSuchAlgorithmException | NoSuchPaddingException | BadPaddingException | IllegalBlockSizeException
-                | InvalidAlgorithmParameterException | InvalidKeyException | IOException e) {
+                | InvalidAlgorithmParameterException | InvalidKeyException e) {
```

Comment:
- remove `IOException` and `Base64` from imports.
- `base64ToBytes()` may throw `IllegalArgumentException` if `data` not in valid base64 format. Don't you want to catch it too?

Reply:
both done